examples/gdp/constrained_layout/cons_layout_model.py

```diff
@@ -148,3 +255,92 @@ def rectangle_in_circle(m, r):
         ]
 
     return m
+
+
+def draw_model(m, title=None):
+    """Draw a model using matplotlib to illustrate what's going on. Pass 'title' kwarg to give chart a title"""
+
+    # matplotlib setup
+    import matplotlib as mpl
+    import matplotlib.pyplot as plt
+
+    fig, ax = plt.subplots(1, 1, figsize=(10, 10))
+    ax.set_xlabel("x")
+    ax.set_ylabel("y")
+
+    # Hardcode these bounds since I'm not sure the best way to do it automatically
+    ax.set_xlim([0, 100])
+    ax.set_ylim([0, 100])
+
+    if title is not None:
+        plt.title(title)
+
+    for c in m.circles:
+        print(
+            f"drawing circle {c}: x={m.circle_x[c]}, y={m.circle_y[c]}, r={m.circle_r[c]}"
+        )
+        # no idea about colors
+        ax.add_patch(
+            mpl.patches.Circle(
+                (m.circle_x[c], m.circle_y[c]),
+                radius=m.circle_r[c],
+                facecolor="#0d98e6",
+                edgecolor="#000000",
+            )
+        )
+        ax.text(
+            m.circle_x[c],
+            m.circle_y[c] + m.circle_r[c] + 1.5,
+            f"Circle {c}",
+            horizontalalignment="center",
+        )
+
+    for r in m.rectangles:
+        print(
+            f"drawing rectangle {r}: x={m.rect_x[r]()}, y={m.rect_y[r]()} (center), L={m.rect_length[r]}, H={m.rect_height[r]}"
```

Comment:
Instead of using `__call__` to get the value of the Vars, you should use Pyomo's `value` function. So for example, replace `m.rect_y[r]()` with `value(m.rect_y[r])`. It will get you better error checking because the implementation of `__call__` will happily return `None` if the Var doesn't have a value, and in this case, getting an error would be better.